src/app/auth/login.ts

```diff
@@ -113,6 +135,136 @@ export const loginWithPhoneCookie = async ({
+  //             c. mongo (account) => update account to level 1
+  //
+  // Scenario 3) Unhappy path - User sells phone and forgets to sweep funds to phone account
+  //             Option 1 - too bad, we can't help them
+  //             Option 2 - create some kind of recovery code process?
+
+  // Scenario 1 - does phone account already exist?
+  const phoneAccount = await UsersRepository().findByPhone(phone)
+  let hasPhoneAccount = true
+  if (phoneAccount instanceof CouldNotFindUserFromPhoneError) {
+    hasPhoneAccount = false
+  } else if (phoneAccount instanceof Error) {
+    return phoneAccount
+  } else if (hasPhoneAccount) {
+    // is there still txns left over on the device account?
+    const deviceWallets = await WalletsRepository().listByAccountId(account.id)
+    if (deviceWallets instanceof Error) return deviceWallets
+    const ledger = LedgerService()
+    let deviceAccountHasBalance = false
+    for (const wallet of deviceWallets) {
+      const balance = await ledger.getWalletBalance(wallet.id)
+      if (balance instanceof Error) return balance
+      if (balance > 0) {
+        deviceAccountHasBalance = true
+      }
+    }
+    if (deviceAccountHasBalance) {
+      // Scenario 1.5 - has txns on device account but phone account exists
+      return new PhoneAccountAlreadyExistsNeedToSweepFundsError()
+    } else {
+      // Scenario 1 - no txns on device account but phone account exists
+      return new PhoneAccountAlreadyExistsError()
+    }
+  }
+
+  // Scenario 2 - Happy Path
+  // a. create kratos account
+  // b. and c. migrate account/user collection in mongo via kratos/registration webhook
+  const kratosToken = await AuthWithDeviceAccountService().upgradeToPhoneSchema({
+    phone,
+    deviceId: account.kratosUserId,
+  })
+  if (kratosToken instanceof Error) return kratosToken
+  return kratosToken
+}
+
+export const loginWithDevice = async ({
+  jwt,
+  ip,
+}: {
+  jwt: string
+  ip: IpAddress
+}): Promise<SessionToken | ApplicationError> => {
+  {
+    const limitOk = await checkFailedLoginAttemptPerIpLimits(ip)
+    if (limitOk instanceof Error) return limitOk
+  }
+
+  const verifiedJwt = await verifyJwt(jwt)
+  if (verifiedJwt instanceof Error) {
+    await rewardFailedLoginAttemptPerIpLimits(ip)
+    return verifiedJwt
+  }
+  const deviceId = verifiedJwt.sub as UserId
+
+  // 1. Does account exist in mongo?
+  const accountExist = await AccountsRepository().findByUserId(deviceId)
+  // 2. If not, then create account in mongo
+  if (accountExist instanceof CouldNotFindAccountFromKratosIdError) {
+    const levelZeroAccountsConfig = getDefaultAccountsConfig()
+    levelZeroAccountsConfig.initialLevel = AccountLevel.Zero
+    const account = await createAccountForDeviceAccount({
+      userId: deviceId,
+      config: levelZeroAccountsConfig,
+    })
+    if (account instanceof Error) return account
+  }
+  return jwt as SessionToken
```

Comment:
do we need to return jwt? could we just return `Success: true` instead. it will make it easier to understand for someone looking into the API what is expected. otherwise someone could assume we get a different JWT here?

At least that my thoughts. wdyt @UncleSamtoshi ?

Reply:
I guess this would break the rollout

Reply:
Right now in the mobile application we use the sessionToken being present as a proxy for success, so we have to keep it to be back backwards compatible, but could migrate over time to `success: true`.